crates/sui-framework/packages/bridge/sources/bridge.move

```diff
@@ -0,0 +1,344 @@
+        self: &mut Bridge,
+        source_chain: u8,
+        bridge_seq_num: u64,
+        ctx: &mut TxContext
+    ): (Option<Coin<T>>, address) {
+        let inner = load_inner_mut(self);
+        let key = message::create_key(source_chain, message_types::token(), bridge_seq_num);
+        if (!linked_table::contains(&inner.bridge_records, key)) {
+            abort EMessageNotFoundInRecords
+        };
+        // retrieve approved bridge message
+        let record = linked_table::borrow_mut(&mut inner.bridge_records, key);
+        // ensure this is a token bridge message
+        assert!(message::message_type(&record.message) == message_types::token(), EUnexpectedMessageType);
+        // Ensure it's signed
+        assert!(option::is_some(&record.verified_signatures), EUnauthorisedClaim);
+
+        // extract token message
+        let token_payload = message::extract_token_bridge_payload(&record.message);
+        // get owner address
+        let owner = address::from_bytes(message::token_target_address(&token_payload));
+
+        // If already claimed, exit early
+        if (record.claimed) {
+            emit(TokenTransferAlreadyClaimed { message_key: key });
+            return (option::none(), owner)
+        };
+
+        let target_chain = message::token_target_chain(&token_payload);
+        // ensure target chain matches self.chain_id
+        assert!(target_chain == inner.chain_id, EUnexpectedChainID);
+
+        // TODO: why do we check validity of the route here? what if inconsistency?
+        // Ensure route is valid
+        // TODO: add unit tests
+        assert!(chain_ids::is_valid_route(source_chain, target_chain), EInvalidBridgeRoute);
+
+        // get owner address
+        let owner = address::from_bytes(message::token_target_address(&token_payload));
+        // check token type
+        assert!(treasury::token_id<T>() == message::token_type(&token_payload), EUnexpectedTokenType);
+        // claim from treasury
+        let token = treasury::mint<T>(&mut inner.treasury, message::token_amount(&token_payload), ctx);
+        // Record changes
+        record.claimed = true;
+        emit(TokenTransferClaimed { message_key: key });
+        (option::some(token), owner)
+    }
+
+    // This function can only be called by the token recipient
+    // Returns None if the token has already been claimed.
+    public fun claim_token<T>(self: &mut Bridge, source_chain: u8, bridge_seq_num: u64, ctx: &mut TxContext): Option<Coin<T>> {
+        let (token, owner) = claim_token_internal<T>(self, source_chain, bridge_seq_num, ctx);
+        // Only token owner can claim the token
+        assert!(tx_context::sender(ctx) == owner, EUnauthorisedClaim);
+        token
+    }
+
+    // This function can be called by anyone to claim and transfer the token to the recipient
+    // If the token has already been claimed, it will return instead of aborting.
+    public fun claim_and_transfer_token<T>(
```

Comment:
Curious on the motivation of this. What would be the case in which we'd want anyone to trigger the claim?

Reply:
this will be used by the bridge client, this function allow it to claim the token for the recipient.